### ISSUE TITLE
Compile to scala-native target

### ISSUE THREAD

**Issue by schrepfler** [frees-io/iota#64]
It would be nice if iota would compile to scala-native as well.

**Comment by andyscott** [comment]
Yep! This really just depends on cats compiling to native. Or Iota getting cross compilation for Scalaz, which already supports native!

**Comment by schrepfler** [comment]
Iota uses full cats?

**Comment by andyscott** [comment]
Iota uses the core FP abstractions from cats as well as Cofree. Internally it uses generalized recursion schemes based on Matryoshka.

**Comment by schrepfler** [comment]
I was looking at using coproduct for case-app to declare many fields which are mutually exclusive (XorK?) and it would be nice if there's support for iota, until then shapeless seems to be the easiest to support as it already has a scala-native port.

**Comment by andyscott** [comment]
I will have Scala Native support coming soon. It will be for the build of Iota that uses Scalaz instead of Cats, as Scalaz already supports Scala Native.
